ogre2/include/ignition/rendering/ogre2/Ogre2ObjectInterface.hh

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#ifndef IGNITION_RENDERING_OGRE2_OGRE2OBJECTINTERFACE_HH_
+#define IGNITION_RENDERING_OGRE2_OGRE2OBJECTINTERFACE_HH_
+
+#include <memory>
+
+#include "ignition/rendering/ogre2/Ogre2Includes.hh"
+
+namespace ignition
+{
+  namespace rendering
+  {
+    inline namespace IGNITION_RENDERING_VERSION_NAMESPACE {
+
+    /// \brief Mixin class to provide direct access to Ogre objects.
+    class Ogre2ObjectInterface
```

Comment:
I think you'll need to add `IGNITION_RENDERING_OGRE2_VISIBLE` here for ignition to export the symbols, i.e.

`class IGNITION_RENDERING_OGRE2_VISIBLE Ogre2ObjectInterface`

Reply:
Added export declarations and provided a default implementation for the `OgreMovableObject` method. Removed empty methods from the camera classes.

I'm not sure that a `const char* _typename` parameter is the best choice, something like the `Ogre::IdString` would be better but I'm not sure of the Ignition equivalent. Given that the function is a placeholder to signal intent, and that any changes to implement the method in camera classes would require a major release in any case, it's probably ok to leave it as is for now.